Pages/Index.cshtml.cs

```diff
@@ -1,100 +1,87 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
-using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
 
-namespace TwilioSMSScheduler.Pages;
+namespace TwilioSmsScheduler.Pages;
 
 public class IndexModel : PageModel
 {
-    private readonly ILogger<IndexModel> _logger;
-    private readonly IConfiguration _configuration;
-    private string clientId, redirectUrl, scope, authBaseUrl, clientSecret, requestTokenBaseUrl = string.Empty;
-    
-    public IndexModel(ILogger<IndexModel> logger, IConfiguration configuration)
+    private readonly ILogger<IndexModel> logger;
+    private readonly IConfiguration configuration;
+    private readonly HttpClient httpClient;
+    private readonly string clientId, redirectUrl, scope, authBaseUrl, clientSecret, requestTokenBaseUrl;
+    private readonly UserCalendarConfiguration userCalendarConfig;
+
+    public IndexModel(ILogger<IndexModel> logger, IConfiguration configuration, HttpClient httpClient)
     {
-        _logger = logger;
-        _configuration = configuration;
-
-        authBaseUrl = _configuration["GoogleAPI:AuthBaseUrl"];
-        clientId = _configuration["GoogleAPI:ClientID"];
-        redirectUrl = _configuration["GoogleAPI:RedirectUrl"];
-        scope = _configuration["GoogleAPI:Scope"];
-        clientSecret = _configuration["GoogleAPI:Secrets"];
-        requestTokenBaseUrl = _configuration["GoogleAPI:RequestTokenUrl"];
+        this.logger = logger;
+        this.configuration = configuration;
+        this.httpClient = httpClient;
+
+        authBaseUrl = this.configuration["GoogleApi:AuthBaseUrl"];
+        clientId = this.configuration["GoogleApi:ClientID"];
+        redirectUrl = this.configuration["GoogleApi:RedirectUrl"];
+        scope = this.configuration["GoogleApi:Scope"];
+        clientSecret = this.configuration["GoogleApi:Secret"];
+        requestTokenBaseUrl = this.configuration["GoogleApi:RequestTokenUrl"];
+        userCalendarConfig = UserCalendarConfiguration.Instance;
+        IsConnected = !string.IsNullOrEmpty(userCalendarConfig.RefreshToken);
+        IsWorkHourSet = !string.IsNullOrEmpty(userCalendarConfig.OpeningTime);
     }
 
-    [BindProperty]
-    public List<string> CheckedDays {get; set;}
+    [BindProperty] public List<string> CheckedDays { get; set; }
 
-    [BindProperty]
-    public List<string> AllDays {get; set;} = new List<string>

```

Comment:
This isn't a form field, so no need to bind.
Also no need to use List since the items won't be modified, no need for a setter either.